application/tests/user_model_test.py

```
@@ -159,6 +162,107 @@ def test_deleting_user_cascades_to_selection(self) -> None:
+        done = threading.Event()
+        errors: List[Exception] = []
+
+        def worker() -> None:
+            with self.app.app_context():
+                try:
+                    # Empty replacement = DELETE only, no INSERT. So the only
+                    # thing that can make this touch (and block on) the User row
+                    # is the method's own FOR UPDATE — which isolates it from the
+                    # FK lock an INSERT would otherwise take.
+                    db.Node_collection().set_user_resource_selection(user_id, [])
+                except Exception as e:
```

Ruff (0.16.1)

[warning] Do not catch blind exception: `Exception`

(BLE001)